File: src/main/java/core/db/test/Supervisor.java
package core.db.test;

import core.db.annotation.Entity;
import core.db.annotation.Id;
import core.db.annotation.Property;

@Entity(table = "Supervisor")
public final class Supervisor {

    @Id(autoincrement = true)
    private long id;

    @Property(name = "first_name")
    private String firstName;

    @Property(name = "last_name")
    private String lastName;

}

File: src/main/java/core/db/test/Supervisor.scala
package core.db.test

import com.j256.ormlite.field.{DatabaseField => Column, ForeignCollectionField => ToMany}
import com.j256.ormlite.table.{DatabaseTable => Table}

object Supervisor {
  def apply(): Supervisor = new Supervisor()
}

@Table(tableName = "supervisor")
final class Supervisor private() {

  @Column(generatedId = true)
  var id: Long = _

  @Column(columnName = "first_name")
  var firstName: String = _

  @Column(columnName = "last_name")
  var lastName: String = _

  override def toString = s"""id=$id, firstName=$firstName, lastName=$lastName"""
}

